src/app/test/test.component.ts

```diff
@@ -299,39 +280,27 @@ export class TestComponent implements OnInit {
     }
   }
 
-  toggleSelectAll(): void {
-    if (this.reports) {
-      if (this.amountOfSelectedReports === this.reports.length) {
-        this.setCheckedForAllReports(false);
-        this.amountOfSelectedReports = 0;
-      } else {
-        this.setCheckedForAllReports(true);
-        this.amountOfSelectedReports = this.reports.length;
-      }
-    }
-  }
-
   setCheckedForAllReports(value: boolean): void {
-    if (this.reports) {
-      for (const report of this.reports) {
-        report.checked = value;
-      }
+    for (const report of this.reports) {
+      report.checked = value;
     }
+    this.refresh();
   }
 
   updatePath(): void {
-    if (this.reports) {
-      const reportIds: number[] = this.helperService.getSelectedIds(this.reports);
-      if (reportIds.length > 0) {
-        const path: string = this.transformPath(this.moveToInputModel.value);
-        const map: UpdatePathSettings = { path: path, action: this.updatePathAction };
-        this.httpService.updatePath(reportIds, this.testReportsService.storageName, map).subscribe({
-          next: () => this.loadData(path),
-          error: () => catchError(this.errorHandler.handleError()),
-        });
-      } else {
-        this.toastService.showWarning('No Report Selected!');
-      }
+    const reportIds: number[] = this.helperService.getSelectedIds(this.reports);
+    if (reportIds.length > 0) {
+      const path: string = this.transformPath(this.moveToInputModel.value);
+      const map: UpdatePathSettings = { path: path, action: this.updatePathAction };
+      this.httpService.updatePath(reportIds, this.testReportsService.storageName, map).subscribe({
+        next: () => this.loadData(path),
+        error: () => catchError(this.errorHandler.handleError()),
+      });
+      this.matches();
+      this.testReportsService.getReports();
+      this.refresh();
```

Comment:
This methods are called before the request is done. Is that correct?

Reply:
you're right. I didn't think of that yet. I added the change.